packages/amplify-category-storage/provider-utils/awscloudformation/service-walkthroughs/s3-walkthrough.js

```diff
@@ -544,6 +544,28 @@ async function updateCfnTemplateWithGroups(context, oldGroupList, newGroupList,
       },
     };
   });
+  // added a new policy for the user group to make action on buckets
+  newGroupList.forEach(group => {
```

Comment:
I don't think this is the right way to go about this. Which "CRUD" operation is this IAM policy tied to? In this implementation, it seems we are adding it to all of the "CRUD" operations.

Reply:
In the above we are adding permissions for objects not on buckets when we do bucketname/*. 
In this to use S3::listbuckets operation , we need IAM policy on bucket.

Reply:
We should associate this policy only if the the selection has "Read" as a part of the CRUD flow.

Reply:
Do you mean this line
```javascript
NewGroupPolicyMap[group].includes('s3:ListBucket') === true
```
Here I am checking if any action is there on bucket , I am adding the policy then.

Reply:
Ah! got it, somehow I missed it.